main/midnight/src/ui/uielement.h

```diff
@@ -211,4 +208,10 @@ static const f32 alpha_zero = 0.0f;
 
 #endif // __UIELEMENT_H_INCLUDED__
 
-
```

Comment:
Reworked this. Moved typedef definition to bottom of file, not protected by #ifndef ... Issue is related to not being able to forward declare typedef the same way we can forward declare classes.

uihelper.h will not compile without typedef being put outside #ifndef block in this header file.

Reply:
That's ok.